infra/Pos/Communication/Relay/Types.hs

```diff
@@ -44,10 +45,17 @@ instance Buildable PropagationMsg where
     build (DataOnlyPM conts) =
         Buildable.build conts
 
--- | Queue of InvMsges which should be propagated.
-type RelayPropagationQueue = TBQueue (Microsecond, PropagationMsg)
+data RelayContext m = RelayContext
```

Comment:
As I wrote further:
> If later you want to support another implementation or add more data you can always expand this type.

So `RelayContext` contains everything you might need for implementation. If at some point need to use something else, you can add it to `RelayContext`. Currently it uses only `DiscoveryContextSum`, so newtype wrapper looks ok to me. If you need `TBQueue`, you can change it to:
```
data RelayContext = RelayContexy DiscoveryContextSum TBQueue
```

>  may also use effects from the application monad, which is not necessarily `IO` nor `MonadIO m => m`.

Yes, it's ok. If you need, for example, `MonadMockable`, you can add it to `WhateverElseConstraintsAreNeeded` I mentioned above and have 
```
  rlyEnqueue :: (Ether.MonadReader' RelayContext m, MonadMockable m) => PropagationMsg -> Maybe NodeId -> m ()
```